datasync/chaindatafetcher/kafka/kafka.go

```diff
@@ -120,9 +125,13 @@ func (k *Kafka) Publish(topic string, data interface{}) error {
 	if err != nil {
 		return err
 	}
+	key := ""
+	if v, ok := data.(IKey); ok {
+		key = v.Key()
+	}
```

Comment:
Is there a valid case when `key == ""`?

Reply:
If the key is not specified (""), the message will be inserted in the Round-Robin way.